src/printer.rs

```diff
@@ -315,17 +310,73 @@ impl<'a> Printer for InteractivePrinter<'a> {
         };
 
         let description = &input.description;
+        let metadata = &input.metadata;
+
+        self.config
+            .header_components
+            .0
+            .iter()
+            .try_for_each(|component| {
+                if self.config.style_components.grid() {
+                    write!(
+                        handle,
+                        "{}{}",
+                        " ".repeat(self.panel_width),
+                        self.colors
+                            .grid
+                            .paint(if self.panel_width > 0 { "│ " } else { "" }),
+                    )?;
+                };
 
-        writeln!(
-            handle,
-            "{}{}{}",
-            description
-                .kind()
-                .map(|kind| format!("{}: ", kind))
-                .unwrap_or_else(|| "".into()),
-            self.colors.filename.paint(description.title()),
-            mode
-        )?;
+                match component {
+                    HeaderComponent::Filename => writeln!(
+                        handle,
+                        "{}{}{}",
+                        description
+                            .kind()
+                            .map(|kind| format!("{}: ", kind))
+                            .unwrap_or_else(|| "".into()),
+                        self.colors.filename.paint(description.title()),
+                        mode
+                    ),
+
+                    HeaderComponent::Size => {
+                        let bsize = metadata
+                            .size
+                            .map(|s| format!("{}", ByteSize(s)))
+                            .unwrap_or("".into());
+                        writeln!(handle, "Size: {}", bsize)
+                    }
+
+                    HeaderComponent::Permissions => writeln!(
+                        handle,
+                        "Permissions: {:o}",
+                        metadata
+                            .permissions
+                            .clone()
+                            .map(|perm| perm.mode)
+                            .unwrap_or(0)
+                    ),
+
+                    HeaderComponent::LastModified => {
+                        let format = format_description::parse(
+                            "[day] [month repr:short] [year] [hour]:[minute]:[second]",
```

Comment:
I'm following `ls -lh` format here, with the difference that I'm including seconds.